pkg/resource/deploy/providers/registry.go

```diff
@@ -30,19 +30,45 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
-// GetProviderVersion fetches and parses a provider version from the given property map. If the version property is not
-// present, this function returns nil.
+var versionKey resource.PropertyKey = "version"
+var pluginDownloadKey resource.PropertyKey = "pluginDownloadURL"
```

Comment:
These can be constants:

```suggestion
const versionKey resource.PropertyKey = "version"
const pluginDownloadKey resource.PropertyKey = "pluginDownloadURL"
```